### ISSUE TITLE
expandAll/collapseAll

### ISSUE THREAD

**Issue by blitz-research** [justinchmura/js-treeview#1]
Hi,

Any chance of adding expandAll/CollapseAll? Or do I just need to do something tricky with the <div>?

Failing that, how about the option to at least create the treeview collapsed?

I'm using it with a large-ish data set, and it works surprisingly well. It's just _too much information_ to have it all expanded at once!

Bye,
Mark

**Comment by justinchmura** [comment]
Hi Mark, I'd be more than happy to add that in. It's probably something I can get in over the weekend if that works for you.

**Comment by blitz-research** [comment]
Yes please!

Also, re: my twitter request for 'node in the callback', I had a poke around and eventually (I suck at JS) found the 'args' object that is sent to callbacks so I can use that, but you might want to doc it!

**Comment by justinchmura** [comment]
Oh I just replied. Guess I should have read this :P the docs can definitely be improved, I agree.

**Comment by justinchmura** [comment]
https://github.com/justinchmura/js-treeview/commit/29c75143464751ff71b40e27be31eacbffca928c

I published version 1.1.1 that includes a `collapseAll` and `expandAll` methods. That should get you some of the way there. I'm currently also adding a way to specify that on initialization.

**Comment by blitz-research** [comment]
Cool, thanks!

I managed to get the treeview to start collapsed by adding this to render after the end of the 'add children' loop:

```
        children.classList.add('hidden');
        expando.textContent = '+';
```

**Comment by justinchmura** [comment]
1.1.2 now includes the option to specify the `expanded` option on a data node. It will now default to collapsed unless specified through the data. Figured that was a better default anyways.

Also, in the event you have this repo forked, I would suggest reforking it. I accidentally did some operations on it that was meant for a different project and it screwed the whole thing up. I had to restart the history.
